pkg/storage/wal/segment.go

```diff
@@ -28,10 +28,12 @@ var (
 	streamSegmentPool = sync.Pool{
 		New: func() interface{} {
 			return &streamSegment{
+				lock:    &sync.Mutex{},
 				entries: make([]*logproto.Entry, 0, 4096),
 			}
 		},
 	}
+	tenantLabel = "__loki_tenant__"
```

Comment:
Had to copy this from `tsdb.TenantLabel` because of a circular import between tsdb & wal via indexWriter.